service/matching/taskListManager.go

```diff
@@ -324,17 +433,12 @@ func (c *taskListManagerImpl) getTask(ctx context.Context) (*getTaskResult, erro
 	}
 
 	select {
-	case task, ok := <-c.taskBuffer:
-		if !ok { // Task list getTasks pump is shutdown
-			c.metricsClient.IncCounter(scope, metrics.PollErrorsCounter)
-			return nil, errPumpClosed
+	case result := <-c.tasksForPoll:
+		if result.syncMatch {
+			c.metricsClient.IncCounter(scope, metrics.PollSuccessWithSyncCounter)
```

Comment:
Remove any counters which are not used anymore.

Reply:
What is not used? We still want a metric for successful poll and whether it is due to sync match or not.